packages/web/components/buttons/back-to-top-button.tsx

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from "react";
+
+import { Button } from "~/components/ui/button";
+
+// use within relative container
+export const BackToTopButton = () => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const toggleVisibility = () => {
+    setIsVisible(window.scrollY > 1);
+  };
+
+  const scrollToTop = () => {
+    window.scrollTo({
+      top: 0,
+      behavior: "smooth",
+    });
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", toggleVisibility);
```

Comment:
What do you think if we use `useSyncExternalStore` instead of `useEffect`?

It is also suggested inside the doc of react: https://react.dev/learn/you-might-not-need-an-effect#subscribing-to-an-external-store